pan3d/ui/globe.py

```diff
@@ -0,0 +1,883 @@
+                v3.VSelect(
+                    placeholder="Color By",
+                    prepend_inner_icon="mdi-format-color-fill",
+                    v_model=(color_by, None),
+                    items=(data_arrays, []),
+                    clearable=True,
+                    hide_details=True,
+                    density="compact",
+                    flat=True,
+                    variant="solo",
+                    max_width=200,
+                )
+
+
+class DataOrigin(CollapsableSection):
+    def __init__(self, load_dataset):
+        super().__init__("Data origin", "show_data_origin", True)
+
+        self.state.load_button_text = "Load"
+        self.state.can_load = True
+        self.state.data_origin_id_to_desc = {}
+
+        with self.content:
+            v3.VSelect(
+                label="Source",
+                v_model=("data_origin_source", "xarray"),
+                items=(
+                    "data_origin_sources",
+                    pan3d_catalogs.list_availables(),
+                ),
+                hide_details=True,
+                density="compact",
+                flat=True,
+                variant="solo",
+            )
+            v3.VDivider()
+            v3.VTextField(
+                placeholder="Location",
+                v_if="['file', 'url'].includes(data_origin_source)",
+                v_model=("data_origin_id", ""),
+                hide_details=True,
+                density="compact",
+                flat=True,
+                variant="solo",
+                append_inner_icon=(
+                    "data_origin_id_error ? 'mdi-file-document-alert-outline' : undefined",
+                ),
+                error=("data_origin_id_error", False),
+            )
+
+            with v3.VTooltip(
+                v_else=True,
+                text=("`${ data_origin_id_to_desc[data_origin_id] }`",),
+            ):
+                with html.Template(v_slot_activator="{ props }"):
+                    v3.VSelect(
+                        v_bind="props",
+                        label="Name",
+                        v_model="data_origin_id",
+                        items=("data_origin_ids", []),
+                        hide_details=True,
+                        density="compact",
+                        flat=True,
+                        variant="solo",
+                    )
+
+            v3.VDivider()
+            v3.VBtn(
+                "{{ load_button_text }}",
+                block=True,
+                classes="text-none",
+                flat=True,
+                density="compact",
+                rounded=0,
+                disabled=("!data_origin_id?.length || !can_load",),
+                color=("can_load ? 'primary': undefined",),
+                click=(
+                    load_dataset,
+                    "[data_origin_source, data_origin_id, data_origin_order]",
+                ),
+            )
+
+
+class DataInformation(CollapsableSection):
```

Comment:
Is that the same as the preview? Can you import and use the one from preview instead?